project/cmake/CMakeLists.txt

```diff
@@ -199,14 +205,22 @@ endif()
 if(NOT WIN32)
   core_optional_dep(OpenGl)
   if(OPENGL_FOUND)
-    core_optional_dep(X ENABLE_X11)
-    core_optional_dep(LibDRM ENABLE_X11)
-    core_optional_dep(XRandR ENABLE_X11)
+    if (ENABLE_MIR)
+      core_optional_dep(Mir ENABLE_MIR)
```

Comment:
core_require_dep please. The dependency is required once the user enables mir.

And please add an option on line 58 for ENABLE_MIR